pinot-core/src/main/java/org/apache/pinot/core/operator/filter/H3IndexFilterOperator.java

```diff
@@ -78,7 +78,15 @@ public H3IndexFilterOperator(IndexSegment segment, QueryContext queryContext, Pr
     assert _h3IndexReader != null;
     int resolution = _h3IndexReader.getH3IndexResolution().getLowestResolution();
     _h3Id = H3Utils.H3_CORE.geoToH3(coordinate.y, coordinate.x, resolution);
-    _edgeLength = H3Utils.H3_CORE.edgeLength(resolution, LengthUnit.m);
+
+    // choose current hexagon's longest edge instead of average for the resolution;
```

Comment:
Is this correct? This is just the max edge length for the center id, but not other hexagons. If the edge length is not all the same, the fundamental assumption doesn't hold, and we need to think for a new algorithm

Reply:
Calculating precise edge length for each ring would most likely defeat any potential performance gains coming from using H3. That's because we could end up calculating distances for up to 30k hexagons even if we don't have any points belonging to them.  For finger-grained precisions (7+) using current edge length should be fine because edge length shouldn't change much within the boundary of 100 rings (or 30k closest hexagons). For coarse-grained precisions, if center is placed near icosahedron edge and distance is large it could end up producing more results because hexagons far from it would be visibly bigger (and we claim them even though we shouldn't ). 

In general - I don't think this application of H3 makes too much sense.
It'd be better to expose library functions such as geoToH3, h3Distance, ... and let users express geo queries in terms of hexagon distances, e.g. 

```sql
SELECT COUNT(*) 
FROM testTable 
WHERE  geoToH3(indexed_point) = geoToH3(ST_Point(-122, 37.5, 1)) 
```
or 
```sql
SELECT COUNT(*) 
FROM testTable 
WHERE  h3Distance( geoToH3(indexed_point),geoToH3(ST_Point(-122, 37.5, 1)) )  < 10
```

Reply:
@yupeng9 Can you please help comment on this? We'll need some guidance on how to use H3 index

Reply:
bumping this @yupeng9 ^